UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs

```diff
@@ -208,5 +209,74 @@ public string GenerateQueryForFss(List<Products> products)
+            string file = Path.Combine(exchangeSetRootPath, fileName);
+            if (!Directory.Exists(exchangeSetRootPath))
+            {
+                Directory.CreateDirectory(exchangeSetRootPath);
+            }
+            string lineToWrite = string.Concat("File date: ", DateTime.UtcNow.ToString("yyyy-MM-dd hh:mm:ssZ"));
+            string secondLineText = string.Empty;
+            HttpResponseMessage httpReadMeFileResponse;
+            httpReadMeFileResponse = await fileShareServiceClient.CallFileShareServiceApi(HttpMethod.Get, payloadJson, accessToken, readMeFilePath);
+            if (httpReadMeFileResponse.IsSuccessStatusCode)
+            {
+                using (Stream stream = await httpReadMeFileResponse.Content.ReadAsStreamAsync())
+                {
+                    using (var outputFileStream = new FileStream(file, FileMode.Create, FileAccess.ReadWrite))
+                    {
+                        stream.CopyTo(outputFileStream);
+                    }
+                    using StreamReader reader = new StreamReader(stream);
+                    secondLineText = GetLine(file, line);
+                }
+                string text = File.ReadAllText(file);
+                text = secondLineText.Length == 0 ? lineToWrite: text.Replace(secondLineText, lineToWrite);
+                File.WriteAllText(file, text);               
+            }
+            else
+                logger.LogInformation(EventIds.ReadMeTextFileIsNotDownloaded.ToEventId(), "Error in downloading readme.txt file for {BatchId}", batchId);
+
+            return true;
+        }
+
+        public async Task<string> SearchReadMeFilePath(string batchId)
+        {
+            string payloadJson = string.Empty;
+            var accessToken = await authTokenProvider.GetManagedIdentityAuthAsync(fileShareServiceConfig.Value.ResourceId);
+            string filePath = string.Empty;           
+            var uri = $"{fileShareServiceConfig.Value.BaseUrl}/batch?$filter={fileShareServiceConfig.Value.ProductType} fileName eq '{fileShareServiceConfig.Value.ReadMeFileName}' and BusinessUnit eq '{fileShareServiceConfig.Value.BusinessUnit}'";
+            HttpResponseMessage httpResponse;
+            httpResponse = await fileShareServiceClient.CallFileShareServiceApi(HttpMethod.Get, payloadJson, accessToken, uri);
+            if (httpResponse.IsSuccessStatusCode)
+            {
+                SearchBatchResponse searchBatchResponse = await SearchBatchResponse(httpResponse);
+                if (searchBatchResponse.Entries.Count > 0)
+                {
+                    var batchResult = searchBatchResponse.Entries.FirstOrDefault();
+                    filePath =  batchResult.Files.FirstOrDefault().Links.Get.Href;                  
+                }
+                else
+                    logger.LogInformation(EventIds.ReadMeTextFileNotFound.ToEventId(), "Readme.txt file not found for {BatchId}", batchId);              
+            }
```

Comment:
can we log non ok response in else like below
               else
                {
                    logger.LogInformation(EventIds.QueryFileShareServiceNonOkResponse.ToEventId(), "Query File share service for readme file with uri {RequestUri} responded with {StatusCode} for {BatchId}", httpResponse.RequestMessage.RequestUri, httpResponse.StatusCode,batchId);
                }

Reply:
Done